onetl/base/base_db_connection.py

```diff
@@ -63,6 +63,23 @@ def validate_columns(cls, connection: BaseDBConnection, columns: list[str] | Non
                 If value is invalid
             """
 
+        @classmethod
+        @abstractmethod
+        def validate_hwm_column(
+            cls,
+            connection: BaseDBConnection,
+            hwm_column: str | tuple[str, str] | Column | None,
```

Comment:
```suggestion
            hwm_column: str | None,
```

Tuple разрезается на hwm_column и hwm_expression, и каждое поле передается в свой валидатор.

Насчет Column VS str, не уверен, что из них лучше - в случае `validate_name` передается `Table`, потому что там отдельные поля db и name, а вот у `Column` есть только `name`, соответственно проще оно не становится.

Возможно стоит все приводить к строке, чтобы уменьшить зависимость интерфейса от текущей реализации.

Reply:
Аннотацию для return тоже нужно поправить